src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: string;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: ShapeType = 'triangle';
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const s = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.round(s * 100) / 100;
+  }
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        `Side a = ${this.a}, side b = ${this.b}, side c = ${this.c} but no side can be equal to zero`,
+      );
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a detailed description in the error message. The message should describe what is specifically wrong, such as 'Triangle side cannot be zero or negative'.